### ISSUE TITLE
Custom build-hooks not supported

### ISSUE THREAD

**Issue by gregoryfischer** [adobe-platform/porter#79]
We have a custom build-hook named `skms_cancel_cmr` defined in our .porter/config. When called in our Jenkins job it throws the following error.

```
21:40:41 INFO[09-06|21:40:41] Hook BEGIN                               porter_version=v2.1.1 cmd=build-hook HookName=skms_cancel_cmr
21:40:41 EROR[09-06|21:40:41] Invalid hook                             porter_version=v2.1.1 cmd=build-hook HookName=skms_cancel_cmr stack="[github.com/adobe-platform/porter/hook/hook.go:104 github.com/adobe-platform/porter/commands/build/hook.go:88 github.com/adobe-platform/porter/vendor/github.com/phylake/go-cli/cli.go:121 porter/main.go:36]"
```

Following the source breadcrumbs it appears custom hooks are not allowed due to this https://github.com/adobe-platform/porter/blob/master/hook/hook.go#L84-L106

**Comment by jfieber** [comment]
@phylake — Is there a reason the hooks are limited to a fixed set?  I have working changes that start with `Hooks` as redefined by `map[string][]Hook` [here](https://github.com/adobe-platform/porter/blob/33b670fe213cc3c2e460b773f54868b0e2b6dee4/conf/conf.go#L64), but that may be at odds with the design intentions.

Pulling back out of this solution and back to the problem, we have a situation where a single logical function needs to be split between a `pre-` and `post-` hook.  In particular, they both need to run regardless of whether the build phase passes or not.  The `post-` operation needs to the pass/fail status of the phase.

The workaround that spawned this issue is to catch the failed phase in the shell script driving porter, and then invoke a custom hook to do the desired `post-` operation for the failure case.  But being able to have a `post-` hook called automatically in _both_ the success and failure case may be a better solution.

**Comment by phylake** [comment]
@jfieber The `porter build hook` command that wasn't kept up to date with hook evolution. It may be as simple as removing the whitelist Greg linked.

In addition to fixing up the command we could add a `run_condition: fail` config to the hook definition that would default to `pass` and optionally run `always`

**Comment by jfieber** [comment]
I'll have a PR for the first option and explore the second, since ultimately it is a better fit for the problem.  Running the manual hook puts logic into the CI system that shouldn't need to be there.

**Comment by phylake** [comment]
Implemented in #97

**Comment by phylake** [comment]
Created #100
